test/TensorFlow/sends_recvs.swift

```diff
@@ -392,3 +392,63 @@ public func testRecvsInALoop() {
   // This one should not be a send.
   _hostOp(a.toHost())
 }
+
+let globalIterator: ResourceHandle =
+  #tfop("Iterator", shared_name: "foo", container: "bar",
+        output_types: [Float.self], output_shapes: [TensorShape()])
+let globalNextIterator: ResourceHandle =
+  #tfop("IteratorGetNextAsOptional", globalIterator,
+        output_types: [Float.self], output_shapes: [TensorShape()])
+
+public func resourceHandlesCanBeSentOrReceived() {
+  // The following constant is a workaround to extend the deabstraction scope so
+  // that a resource handle is not an input argument to the tensor program. This
+  // will go away when we convert arg tensors to Swift->TF sends.
+  let _ = Tensor<Float>(1.0)
+  // expected-warning @+2 {{value implicitly copied to the accelerator}}
+  let _: VariantHandle =
+    #tfop("IteratorGetNextAsOptional", globalIterator,
+    output_types: [Float.self], output_shapes: [TensorShape()])
+  // expected-warning @+2 {{value implicitly copied to the host}}
+  let localIterator: ResourceHandle =
+    #tfop("Iterator", shared_name: "foo", container: "bar",
+          output_types: [Float.self], output_shapes: [TensorShape()])
+  _hostOp(localIterator) // expected-note {{value used here}}
```

Comment:
is it valid to print a resource tensor? i suspect runtime will crash. if so, we should find a way to reject the program in compiler (pls add a bug/TODO to track that if it makes sense).

Reply:
It just prints "TensorFlow.ResourceHandle" or "TensorFlow.VariantHandle". We should probably make it print more meaningful information. 

In any case, I don't think we should reject it in the compiler as this is not a type error?

Reply:
SG

Reply:
Yep, works with debug log as well. I believe debugLog uses print underneath anyway?